test/networks.js

```diff
@@ -40,6 +40,25 @@ describe('Networks', function() {
       }
     }
   });
+  
+  it('should not set a network map for an undefined value', function() {
+    var custom = {
+      name: 'somenet',
+      pubkeyhash: 0x13,
+      privatekey: 0x93,
+      scripthash: 0x11,
+      xpubkey: 0x0278b20f,
+      xprivkey: 0x0278ade5,
+      networkMagic: 0xe7beb4d5,
+      port: 20008,
+      dnsSeeds: [
+        'somenet.localhost'
+      ]
+    };
+    networks.add(custom);
+    var network = networks.get(undefined);
+    should.not.exist(network);
```

Comment:
I see. I'll merge then

On Tue, Mar 24, 2015, 20:07 Braydon Fuller <notifications@github.com> wrote:

> In test/networks.js
> <https://github.com/bitpay/bitcore/pull/1154#discussion_r27082474>:
>
> > +    var custom = {
> > +      name: 'somenet',
> > +      pubkeyhash: 0x13,
> > +      privatekey: 0x93,
> > +      scripthash: 0x11,
> > +      xpubkey: 0x0278b20f,
> > +      xprivkey: 0x0278ade5,
> > +      networkMagic: 0xe7beb4d5,
> > +      port: 20008,
> > +      dnsSeeds: [
> > +        'somenet.localhost'
> > +      ]
> > +    };
> > +    networks.add(custom);
> > +    var network = networks.get(undefined);
> > +    should.not.exist(network);
>
> I don't think so currently, since there is code that is written like this:
> var network = Networks.get(argument) || Networks.defaultNetwork;
> If argument is undefined, undefined is returned and then is set to the
> default.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/bitpay/bitcore/pull/1154/files#r27082474>.
>